src/power_grid_model/_utils.py

```diff
@@ -177,7 +179,7 @@ def split_numpy_array_in_batches(data: DenseBatchArray | SingleArray, component:
     )
 
 
-def split_sparse_batches_in_batches(batch_data: SparseBatchArray, component: ComponentType) -> list[np.ndarray]:
+def split_sparse_batches_in_batches(batch_data: SparseBatchData, component: ComponentType) -> list[SingleArray]:
```

Comment:
By allowing this type, line 205 might give typeerror.

Reply:
good one.